Mathlib/NumberTheory/Cyclotomic/Basic.lean

```diff
@@ -578,10 +579,12 @@ instance algebraBase : Algebra A (CyclotomicRing n A K) :=
 -- but there is at `reducible_and_instances` #10906
 example {n : ℕ+} : CyclotomicRing.algebraBase n ℤ ℚ = Ring.toIntAlgebra _ := rfl
 
-instance : NoZeroSMulDivisors A (CyclotomicRing n A K) :=
+instance [NoZeroSMulDivisors A (CyclotomicField n K)] :
```

Comment:
I'm confused by this one: it seems that you're adding a typeclass assumption, but that you haven't deleted it in this file. Same thing below.

Reply:
Oops, this had previously been supplied indirectly via the `[IsFractionRing A K]` deleted above, but I guess I just added the typeclass that Lean reported was missing, rather than the stronger deleted assumption.

I'll change these back to `[IsFractionRing A K]`, as it seems unlikely that any added generality is actually useful or helpful here!!